### PR TITLE
Add support for defining host's macros

### PR DESCRIPTION
**Pull request by zv0n**
#### Pull Request (PR) description
This commit adds support for defining custom macros for zabbix hosts

### CODE REVIEW

File: manifests/agent.pp [hunk 1]
```diff
@@ -239,6 +242,7 @@
   Variant[String[1],Array[String[1]]] $zbx_groups = $zabbix::params::agent_zbx_groups,
   $zbx_group_create                               = $zabbix::params::agent_zbx_group_create,
   $zbx_templates                                  = $zabbix::params::agent_zbx_templates,
+  $zbx_macros                                     = $zabbix::params::agent_zbx_macros,
```
Comment (bastelfreak) [inline manifests/agent.pp:245]: please add a datatype for the new parameter

**Comment by bastelfreak** [comment]
thanks for the work  @zv0n. can you please rebase?

**Comment by zv0n** [comment]
Hi,
I've rebased and added the datatype

**Comment by bastelfreak** [comment]
@zv0n can you please fix the failing travis job?

**Comment by zv0n** [comment]
Hi,
I've fixed the jobs

File: manifests/agent.pp [hunk 2]
```diff
@@ -249,6 +252,7 @@
   Variant[String[1],Array[String[1]]] $zbx_groups = $zabbix::params::agent_zbx_groups,
   $zbx_group_create                               = $zabbix::params::agent_zbx_group_create,
   $zbx_templates                                  = $zabbix::params::agent_zbx_templates,
+  Optional[Array[Hash]] $zbx_macros               = $zabbix::params::agent_zbx_macros,
```
Comment (bastelfreak) [inline manifests/agent.pp:255]: for arrays we try to use `[]` as default, or `{}` for hashes. In that case we won't need the `Optional` part. I think that would be possible here was well?

We've more of our review guidelines available at https://voxpupuli.org/docs/reviewing_pr/

File: manifests/agent.pp [hunk 3]
```diff
@@ -249,6 +252,7 @@
   Variant[String[1],Array[String[1]]] $zbx_groups = $zabbix::params::agent_zbx_groups,
   $zbx_group_create                               = $zabbix::params::agent_zbx_group_create,
   $zbx_templates                                  = $zabbix::params::agent_zbx_templates,
+  Array[Hash] $zbx_macros                         = $zabbix::params::agent_zbx_macros,
```
Comment (bastelfreak) [inline manifests/agent.pp:255]:
```suggestion
  Array[Hash] $zbx_macros                         = [],
```

File: manifests/params.pp [hunk 4]
```diff
@@ -317,6 +317,7 @@
   $agent_zbx_templates                      = [
     'Template OS Linux',
     'Template App SSH Service']
+  $agent_zbx_macros                         = undef
```
Comment (bastelfreak) [inline manifests/params.pp:320]: you can remove this line

**Comment by zv0n** [comment]
Looks like the job failed because there was a problem problem when connecting to the container, should I do something or is the pull request ok now?

**Comment by bastelfreak** [comment]
I restarted the job

**Pull request merged**